optimize/backend/src/main/java/io/camunda/optimize/tomcat/ExternalApiRewriteFilter.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.optimize.tomcat;
+
+import jakarta.servlet.Filter;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.RequestDispatcher;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.ServletRequest;
+import jakarta.servlet.ServletResponse;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import org.slf4j.Logger;
+
+public class ExternalApiRewriteFilter implements Filter {
+
+  private static final String EXTERNAL_API_PREFIX = "/external/api";
+  private static final String API_EXTERNAL_PREDIX = "/api/external";
+  private static final Logger LOG =
+      org.slf4j.LoggerFactory.getLogger(ExternalApiRewriteFilter.class);
+
+  public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+      throws IOException, ServletException {
+    HttpServletRequest httpRequest = (HttpServletRequest) request;
+    HttpServletResponse httpResponse = (HttpServletResponse) response;
+
+    String requestURI = httpRequest.getRequestURI();
+
+    if (requestURI.contains(EXTERNAL_API_PREFIX)) {
```

Comment:
In the previous case, we had a regular expression matching.

I got a bit defensive with these because in a separate case, that didn't work as expected anymore (didn't investigate why). I am trying to keep things simpler, when more complex code is not necessary. Let me know if you want me to restore the previous pattern matching.

In this case, we can be more specific and I will do it, but basically it's the case where the URL starts with `/external/api/`. I can change contains with startsWith.

Reply:
replaced contains in favor of startsWith

Reply:
Happy with startsWith, as long as this works with the context path too as discussed 👍